Upgrade v1/v2v3 project activity CSV exports

filipvvv opened the pull request:
## What does this PR do and why?

Add redemptions, payouts, additions to balance, and participants to the CSV export modal on v2/v3 project pages. Consolidate into one modal (instead of bringing up a second modal for block number). I'll do v1 projects some time soon.

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96905094/205792818-5778646c-4b78-4dee-9f21-8538af77b6a3.mov

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

tomquirk commented:
@filipvvv sir, thank you for this PR and for taking the initiative here! Will be so great to have richer export tools. Great great work 👏 

My only suggestion ATM would be to reduce the size of this file by splitting the code up across multiple files. This will help with maintainability. One approach might be to create components for each button (for example, `DownloadPaymentsCsvButton.tsx`), and keep the related download utility function (for example `downloadPayments(...)`) in each component. Alternatively, we could move the download helper functions to their own file.

Let me know what you think or if you have any questions.

filipvvv commented:
> @filipvvv sir, thank you for this PR and for taking the initiative here! Will be so great to have richer export tools. Great great work clap
> 
> My only suggestion ATM would be to reduce the size of this file by splitting the code up across multiple files. This will help with maintainability. One approach might be to create components for each button (for example, `DownloadPaymentsCsvButton.tsx`), and keep the related download helper function (for example `downloadPayments(...)`) in each component. Alternatively, we could move the download helper functions to their own file.
> 
> Let me know what you think or if you have any questions.

SGTM! Did this, implemented for v1, and added the USD amounts from #2708 (merged into this PR). Also rebased with main.

filipvvv commented:
Let me know if I should re-do this PR sans the rebase.

tomquirk commented:
Hey @filipvvv I think something has gone a bit skewiff with the rebase - no worries, I will take a look and force-push a clean rebase in a sec!

tomquirk commented:
@filipvvv okay! Think we're good now. I made some super minor changes [here](https://github.com/jbx-protocol/juice-interface/pull/2632/commits/c007eaafc6acf9a695535dffee9a27267e972f78), just make the modal markup more consistent with our commonly used patterns in the codebase.

I also merged this PR from Peri prior: https://github.com/jbx-protocol/juice-interface/pull/2708, which included some of the USD subgraph additions.

All in all, I've tested all the downloads and they appear to be working as expected. Lets ship it!!! Thanks again for working on this.

filipvvv commented on src/utils/buildActivityCsvs.ts:
```diff
@@ -0,0 +1,517 @@
+    additions.forEach(a => {
+      let date = new Date((a.timestamp ?? 0) * 1000).toUTCString()
+
+      if (date.includes(', ')) date = date.split(', ')[1]
```
> **suggestion** looks like this code can be consolidated into a single util to reduce duplication. WDYT?

Up to you! There are a few things which could be broken up into utils but it would make the code a bit harder to follow. Worth keeping consolidated in the functions IMO.

filipvvv commented on src/utils/buildActivityCsvs.ts:
```diff
@@ -0,0 +1,517 @@
+      if (date.includes(', ')) date = date.split(', ')[1]
+
+      let beneficiary = p.modBeneficiary
+      if (p.modProjectId) beneficiary = `v1 project ${p.modProjectId}`
```
> **suggestion (non-blocking)**

Yes, very good! Could even move into the rows.push. Maybe I'll look over this again some time soon — feel free to make changes if you feel inclined to do so.
